src/app/[id]/edit/page.tsx

```diff
@@ -0,0 +1,283 @@
+'use client';
+
+import React, {useEffect, useState} from 'react';
+import {title} from '@prisma/client';
+import {getLocalTitle, postLocalTitle} from '@/services/local.data';
+import {fetchNewspaperTitleFromCatalog} from '@/services/catalog.data';
+import {CatalogTitle} from '@/models/CatalogTitle';
+import {Field, Form, Formik} from 'formik';
+import {useRouter} from 'next/navigation';
+import {Button} from '@nextui-org/button';
+
+export default function Page({params}: { params: { id: string } }) {
+  const router = useRouter();
+  const [catalogTitle, setCatalogTitle] = useState<CatalogTitle>();
+  const [localTitle, setLocalTitle] = useState<title>();
+  const [saveMessageIsVisible, setSaveMessageIsVisible] = useState<boolean>(false);
+
+  useEffect(() => {
+    void fetchNewspaperTitleFromCatalog(params.id).then((data: CatalogTitle) => setCatalogTitle(data));
+  }, [params]);
+
+  useEffect(() => {
+    void getLocalTitle(params.id)
+      .then((data: title) => setLocalTitle(data))
+      .catch((e: string) => {
+        if (e.toLowerCase().includes('not found')) {
```

Comment:
Anbefaler å definere en ny typ Error, istedenfor å lese feilmeldingen. Da kan vi ha:

```suggestion
      .catch((e: Error) => {
        if (e instance of NotFoundError) {
```
 Endringene i `local.data.ts`:

```
export class NotFoundError extends Error {}
...
case 404:
    return Promise.reject(new NotFoundError('Title not found'));
  default:
    return Promise.reject(new Error('Failed to fetch title'));
```

Reply:
Oppdatert!